viadot/sources/outlook.py

```diff
@@ -92,66 +99,71 @@ def to_df(self) -> pd.DataFrame:
             pd.DataFrame: the DataFrame with time range
         """
         data = []
-
-        while True:
-            try:
-                message = next(self.mailbox_messages)
-                received_time = message.received
-                date_time_str = str(received_time)
-                date_time_string = date_time_str[0:19]
-                date_obj = datetime.datetime.strptime(
-                    date_time_string, "%Y-%m-%d %H:%M:%S"
-                )
-                if (
-                    date_obj < self.date_range_start_time
-                    or date_obj > self.date_range_end_time
-                ):
-                    continue
-                else:
-                    fetched = message.to_api_data()
-                    try:
-                        sender_mail = fetched["from"]["emailAddress"]["address"]
-                        reciver_list = fetched.get("toRecipients")
-                        recivers = " "
-                        if reciver_list is not None:
-                            recivers = ", ".join(
-                                reciver["emailAddress"]["address"]
-                                for reciver in reciver_list
-                            )
-
-                        categories = " "
-                        if message.categories is not None:
-                            categories = ", ".join(
-                                categories for categories in message.categories
+        mailbox_generators_list = []
+        for m in self.mailbox_folders:
+            base_str = f"self.mailbox_obj.{m}_folder().get_messages({self.limit})"
+            mailbox_generators_list.append(eval(base_str))
+
+        for mailbox_generator in mailbox_generators_list:
+            while True:
+                try:
+                    message = next(mailbox_generator)
+                    received_time = message.received
+                    date_time_str = str(received_time)
+                    date_time_string = date_time_str[0:19]
```

Comment:
Maybe change the name of variable. It's okay, but it's almost the same as `date_time_str`. I would suggest sth like `date_time_stripped` or `date_time_trimmed`.